<?xml version="1.0" encoding="UTF-8"?>

<launch>

	<arg name="time_step" default="0.5" />
	<arg name="rolename" />
	<arg name="steps" default="20" />
	<arg name="distance" default="4.0" />


	<group ns="$(arg rolename)">

		<!-- start service call for game node  -->
		<node name="restart_call" pkg="carla_circle" type="game_start_client.py" output="screen" />

		<!-- "perception module", acutally just filtering out cars that are relevant -->
		<node name="game_ado_car_filter" pkg="carla_circle" type="game_ado_vehicle_filter.py" output="screen">
			<param name="rolename" value="$(arg rolename)" />
		</node>

		<!-- this node works for game planner to publish current track center and track width for two cars -->
		<node name="map_updater" pkg="carla_circle" type="map_updater.py" output="screen">
			<param name="steps" value="$(arg steps)" />
			<param name="distance" value="$(arg distance)" />
			<param name="update_frequency" value="10.0" />
			<param name="exit_time" value="300" />
		</node>
		<!-- path server updates path according to carla world map -->
		<node name="path_server" pkg="carla_circle" type="path_server.py" output="screen">
			<param name="steps" value="$(arg steps)" />
			<param name="distance" value="$(arg distance)" />
		</node>

		<!-- controller module  -->
		<node name="controller" pkg="carla_circle" type="controller.py" output="screen">
			<param name="rolename" value="$(arg rolename)" />
			<param name="time_step" value="$(arg time_step)" />
			<param name="ctrl_freq" value="100" />
			<param name="str_prop" value="0.5" />
			<param name="plan_steps" value="10" />
		</node>

		<node name="viz_node" pkg="carla_circle" type="visualization_node.py" output="screen" />

		<node name="autonomous_agent" pkg="carla_circle" type="manual_control.py" args="--positionx 50.0 --rolename stanford_ego --color '255,200,200'"	output="screen" />
	</group>

</launch>